packages/base-contract/src/index.ts

```diff
@@ -1,40 +1,31 @@
 import { ContractAbi, DataItem, TxData, TxDataPayable } from '@0xproject/types';
+import { BigNumber } from '@0xproject/utils';
 import { Web3Wrapper } from '@0xproject/web3-wrapper';
 import * as ethersContracts from 'ethers-contracts';
 import * as _ from 'lodash';
 
+import { formatABIDataItem } from './utils';
+
 export class BaseContract {
     protected _ethersInterface: ethersContracts.Interface;
     protected _web3Wrapper: Web3Wrapper;
     public abi: ContractAbi;
     public address: string;
-    protected static _transformABIData(
+    protected static _formatABIDataItemList(
         abis: DataItem[],
         values: any[],
-        transformation: (type: string, value: any) => any,
+        formatter: (type: string, value: any) => any,
     ): any {
-        return _.map(values, (value: any, i: number) =>
-            BaseContract._transformTypedData(abis[i].type, value, transformation),
-        );
+        return _.map(values, (value: any, i: number) => formatABIDataItem(abis[i], value, formatter));
     }
     protected static _lowercaseAddress(type: string, value: string): string {
         return type === 'address' ? value.toLowerCase() : value;
     }
-    protected static _bigNumberToString(type: string, value: string): string {
-        return _.isObject(value) && (value as any).isBigNumber ? value.toString() : value;
+    protected static _bigNumberToString(type: string, value: any): any {
```

Comment:
Why not `value: BigNumber): BigNumber`?

Reply:
Because the real semantic of that function is: "Get a value and if it's a BigNumber - return a string. Otherwise - noop." With TS 2.7 I can't express such a logic in a type system. With TS 2.8 cond types that's possible. I'll change that as soon as we migrate.